diff --git a/Makefile.am b/Makefile.am
index 6d4cd32..0dee636 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -21,10 +21,7 @@
 #  TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
 #  PERFORMANCE OF THIS SOFTWARE.
 
-SUBDIRS=fontconfig fc-case fc-lang src \
-	fc-cache fc-cat fc-conflist fc-list fc-match \
-	fc-pattern fc-query fc-scan fc-validate conf.d \
-	its po po-conf test
+SUBDIRS=fontconfig src
 if ENABLE_DOCS
 SUBDIRS += doc
 endif
